test/test_stdout.rb

```diff
@@ -18,4 +18,9 @@ def test_default_list_output
       assert_match(/.*#{spec_key}.*/, @stdout.string)
     end
   end
+
+  def test_ubuntu_host_generation_output
```

Comment:
Ah yeah didn't notice the `$` vs `@`, or the `setup` code.

I'm not sure that `setup` runs between each test though; I think it's just once at the beginning.
I would be worried about cross-test contamination.

Since our test is about redirecting the stream, I would expect it to explicitly:

1. Redirect `$stderr` to a new `StringIO`
2. Call `CLI#execute!`
3. Assert the stream contents

You might have to put the old stream back at the end of the test, or instead make `setup` run for every test rather than once.

[META: I'm probably going to be cleaning some of this up very soon when I get rid of the warning like I mentioned in that ticket, so don't worry too much about it being perfect.  I'm more interested in using this as an opportunity to share my thoughts on testing than I am about making this particular test better.]